src/ui/public/vis/request_handlers/courier.js

```diff
@@ -35,15 +37,22 @@ const CourierRequestHandlerProvider = function (Private, courier, timefilter) {
             };
 
             searchSource.rawResponse = resp;
-            resolve(resp);
+
+            resolve(_.cloneDeep(resp));
           }).catch(e => reject(e));
 
           courier.fetch();
         } else {
-          resolve(searchSource.rawResponse);
+          resolve(_.cloneDeep(searchSource.rawResponse));
         }
-
-
+      }).then(async resp => {
+        for (let i = 0; i < vis.aggs.length; i++) {
+          const agg = vis.aggs[i];
+          if (!agg.type || !agg.type.postFlightRequest) continue;
```

Comment:
I think we should rather be as failsafe as possible and check if the function exists here before calling it, instead of relying on some complete other place always attaching the identity function for us.